### DISCUSSION TITLE
feat: add a feature to an existing application scaffolded with bati

### DISCUSSION THREAD

**Discussion by aheissenberger** [vikejs/bati#367] [Misc]
Use bati to add a feature to an existing application which was scaffolded by bati.

**Example:**

1. create the application:
`pnpm create bati  --react my-project`

2. add a feature:
`pnpm create bati --tailwindcss`

**Solution:**
1. requires a bati config file oder a section in packages.json with the information about the features which have been installed when bati was creating the project. 
2. Based on this list, bati will merge the old and new feature list. 
3. Creates a new project in a temp folder. 
4. Provides the user with a prompt to accept diffs when new files differ from existing files. 
5. Templates which support direct manipulation (e.g. `$vite.config.ts.ts`) should use the existing files as a base (Problem: addVitePlugin will fail if import exists)
6. Update the list in package.json after a successful install of new features.

**Currently behavior:**
* bati will only write into an existing project with the option `--force`
* requires the knowledge of all installed features as this information is only as text available in the README.md
* overwrites changes to files which have been modified after the creation of the project

**Use Cases:**
* add Database
* add Deployment Adapter
* add Auth

**Problems:**
* files of already installed features can change based on the new feature. This will require to ask the User to accept the changes

**Comment by magne4000** [comment]
That is a nice idea, and I already considered it. But I don't see this realistically happening.

Adding features to an existing project could be straightforward, with just a new bati state file, if it hasn't been modified. But one can create a new project with the added feature as a simple workaround.
If the project has been modified, it's another story. I foresee that most users will have many conflicts, and most of them will be non trivial.
And this is without talking of the fact that Bati's codebase can be refactored, files moved around, etc. It'll be impossible to keep track of all this.

That being said, the possibility of adding or removing a feature is part of our [Stem Vision](https://github.com/magne4000/universal-middleware/discussions/5), and it will be compaitble with the Vike ecosystem once we'll start working on it.
